jvb/src/main/kotlin/org/jitsi/videobridge/cc/allocation/SingleSourceAllocation.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright @ 2020 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge.cc.allocation
+
+import org.jitsi.nlj.RtpLayerDesc
+import org.jitsi.videobridge.cc.config.BitrateControllerConfig.Companion.onstagePreferredFramerate
+import org.jitsi.videobridge.cc.config.BitrateControllerConfig.Companion.onstagePreferredHeightPx
+import java.lang.Integer.max
+
+/**
+ * Gets the "preferred" height and frame rate based on the constraints signaled from the received.
+ *
+ * For participants with sufficient maxHeight we favor frame rate over resolution. We consider all
+ * temporal layers for resolutions lower than the preferred, but for resolutions >= preferred, we only
+ * consider frame rates at least as high as the preferred. In practice this means we consider
+ * 180p/7.5fps, 180p/15fps, 180p/30fps, 360p/30fps and 720p/30fps.
+ */
+fun getPreferred(constraints: VideoConstraints): Pair<Int, Double> {
+    return if (constraints.maxHeight > 180) {
```

Comment:
I don't mind extracting a constant, but I don't understand the rationale for this in the first place, so it would be a constant for "min height for 'preferred' to be used", which doesn't seem useful.